google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerCloudMonitoringExporter.java

```diff
@@ -0,0 +1,246 @@
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.threeten.bp.Duration;
+
+/**
+ * Spanner Cloud Monitoring OpenTelemetry Exporter.
+ *
+ * <p>The exporter will look for all spanner owned metrics under spanner.googleapis.com
+ * instrumentation scope and upload it via the Google Cloud Monitoring API.
+ */
+class SpannerCloudMonitoringExporter implements MetricExporter {
+
+  private static final Logger logger =
+      Logger.getLogger(SpannerCloudMonitoringExporter.class.getName());
+
+  // This system property can be used to override the monitoring endpoint
+  // to a different environment. It's meant for internal testing only.
+  private static final String MONITORING_ENDPOINT =
+      MoreObjects.firstNonNull(
+          System.getProperty("spanner.test-monitoring-endpoint"),
+          MetricServiceSettings.getDefaultEndpoint());
+
+  // This the quota limit from Cloud Monitoring. More details in
+  // https://cloud.google.com/monitoring/quotas#custom_metrics_quotas.
+  private static final int EXPORT_BATCH_SIZE_LIMIT = 200;
+  private final AtomicBoolean spannerExportFailureLogged = new AtomicBoolean(false);
+  private CompletableResultCode lastExportCode;
+  private final MetricServiceClient client;
+  private final String spannerProjectId;
+  private final AtomicBoolean isShutdown = new AtomicBoolean(false);
+
+  static SpannerCloudMonitoringExporter create(String projectId, @Nullable Credentials credentials)
+      throws IOException {
+    MetricServiceSettings.Builder settingsBuilder = MetricServiceSettings.newBuilder();
+    CredentialsProvider credentialsProvider;
+    if (credentials == null) {
+      credentialsProvider = NoCredentialsProvider.create();
+    } else {
+      credentialsProvider = FixedCredentialsProvider.create(credentials);
+    }
+    settingsBuilder.setCredentialsProvider(credentialsProvider);
+    settingsBuilder.setEndpoint(MONITORING_ENDPOINT);
+
+    org.threeten.bp.Duration timeout = Duration.ofMinutes(1);
+    // TODO: createServiceTimeSeries needs special handling if the request failed. Leaving
+    // it as not retried for now.
+    settingsBuilder.createServiceTimeSeriesSettings().setSimpleTimeoutNoRetries(timeout);
+
+    return new SpannerCloudMonitoringExporter(
+        projectId, MetricServiceClient.create(settingsBuilder.build()));
+  }
+
+  @VisibleForTesting
+  SpannerCloudMonitoringExporter(String projectId, MetricServiceClient client) {
+    this.client = client;
+    this.spannerProjectId = projectId;
+  }
+
+  @Override
+  public CompletableResultCode export(Collection<MetricData> collection) {
+    if (isShutdown.get()) {
+      logger.log(Level.WARNING, "Exporter is shutting down");
+      return CompletableResultCode.ofFailure();
+    }
+
+    lastExportCode = exportSpannerClientMetrics(collection);
```

Comment:
nit (this makes it a bit easier to quickly see that we are setting an instance field):
```suggestion
    this.lastExportCode = exportSpannerClientMetrics(collection);
```